apps/admin_app/lib/admin_app_web/controllers/prototype_controller.ex

```diff
@@ -0,0 +1,95 @@
+defmodule AdminAppWeb.PrototypeController do
+  use AdminAppWeb, :controller
+
+  alias Snitch.Data.Schema.ProductPrototype, as: PrototypeSchema
+  alias Snitch.Data.Model.ProductPrototype, as: PrototypeModel
+  alias Snitch.Data.Model.{Property, VariationTheme}
+  alias Snitch.Domain.Taxonomy
+
+  plug(:load_resources when action in [:new, :edit, :update, :create])
+
+  def index(conn, _params) do
+    prototypes = PrototypeModel.get_all()
+    render(conn, "index.html", prototypes: prototypes)
+  end
+
+  def new(conn, _params) do
+    changeset = PrototypeSchema.create_changeset(%PrototypeSchema{}, %{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, %{"product_prototype" => params}) do
+    case PrototypeModel.create(params) do
+      {:ok, _} ->
+        prototypes = PrototypeModel.get_all()
+        render(conn, "index.html", prototypes: prototypes)
+
+      {:error, changeset} ->
+        render(conn, "new.html", changeset: %{changeset | action: :new})
+    end
+  end
+
+  def edit(conn, %{"id" => id}) do
+    with {id, _} <- Integer.parse(id),
```

Comment:
Ecto provides `id` casting. I think it is not necessary to parse the id using `Integer.parse`. There was an issue raised in 2014 on this. I think they are providing now.
https://github.com/elixir-ecto/ecto/issues/282

Reply:
Yes Ecto does the casting.
Query helper that we use in models only accepts `non_integer` values. Will make change to accept `binary` values to, so we don't have to parse the id each time.